### ISSUE TITLE
Async usage is broken as of 0.29.0 for client.messages.create

### ISSUE THREAD

**Issue by akashkashyap** [anthropics/anthropic-sdk-python#610]
**tl;dr AsyncAnthropic is broken as of 0.29.0 for messages.create. If I roll back to to the previous release 0.28.1, it does not have this issue. I think [this commit](https://github.com/anthropics/anthropic-sdk-python/commit/3c2b75fac662e48effc8ec032266d966e548007d) might be a possible culprit but I'm not sure.**

Observed while running a test suite today. After running 2 tests that make 2 async client.messages.create calls (successfully), the 3rd call (in the 3rd test) hangs every time. Note, the first 2 calls don't include tools and the 3rd call does. The tests all use the async client, but they run one after another (very quickly, minimal gap between each one).

This is consistently reproducible with 0.29.0. The same flow doesn't hang at all when using the previous release 0.28.1.

I've included the trace that dumps when I kill the hanging test below, in case that has any hints towards a fix.

> `File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/anthropic/resources/messages.py", line 1866, in create
>     return await self._post(
>            ^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/anthropic/_base_client.py", line 1804, in post
>     return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/anthropic/_base_client.py", line 1502, in request
>     return await self._request(
>            ^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/anthropic/_base_client.py", line 1536, in _request
>     response = await self._client.send(
>                ^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1661, in send
>     response = await self._send_handling_auth(
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1689, in _send_handling_auth
>     response = await self._send_handling_redirects(
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1726, in _send_handling_redirects
>     response = await self._send_single_request(request)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1763, in _send_single_request
>     response = await transport.handle_async_request(request)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 373, in handle_async_request
>     resp = await self._pool.handle_async_request(req)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpcore/_async/connection_pool.py", line 216, in handle_async_request
>     raise exc from None
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpcore/_async/connection_pool.py", line 196, in handle_async_request
>     response = await connection.handle_async_request(
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpcore/_async/connection.py", line 99, in handle_async_request
>     raise exc
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpcore/_async/connection.py", line 76, in handle_async_request
>     stream = await self._connect(request)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpcore/_async/connection.py", line 122, in _connect
>     stream = await self._network_backend.connect_tcp(**kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpcore/_backends/auto.py", line 30, in connect_tcp
>     return await self._backend.connect_tcp(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/httpcore/_backends/anyio.py", line 116, in connect_tcp
>     stream: anyio.abc.ByteStream = await anyio.connect_tcp(
>                                    ^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/anyio/_core/_sockets.py", line 214, in connect_tcp
>     async with create_task_group() as tg:
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/anyio/_backends/_asyncio.py", line 690, in __aexit__
>     raise cancelled_exc_while_waiting_tasks
>   File "/Users/user/workspace/project/.venv/lib/python3.12/site-packages/anyio/_backends/_asyncio.py", line 670, in __aexit__
>     await asyncio.wait(self._tasks)
>   File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/tasks.py", line 464, in wait
>     return await _wait(fs, timeout, return_when, loop)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/tasks.py", line 550, in _wait
>     await waiter
> asyncio.exceptions.CancelledError`

**Comment by RobertCraigie** [comment]
Can you share a reproduction script?

**Comment by RobertCraigie** [comment]
Hi @akashkashyap can you share any more details?

A script like this works as expected for me
```py
import asyncio
from anthropic import AsyncAnthropic

async def main() -> None:
    client = AsyncAnthropic()

    for i in range(10):
        await make_request(client)
        print(f"success {i}")

async def make_request(client: AsyncAnthropic) -> None:
    await client.messages.create(
        max_tokens=1024,
        messages=[
            {
                "role": "user",
                "content": "Hello, Claude",
            }
        ],
        model="claude-3-opus-20240229",
    )

asyncio.run(main())
```

**Comment by akashkashyap** [comment]
Hi @RobertCraigie, thanks for the quick follow-up.

I'm unable to get it to repro consistently this morning with the test suite I described in the original post. FWIW, adding tools in the request might help repro because the last call in the test suite included tools. Feel free to resolve this for now, I can open a new issue again if/when I see it.
